app/assets/javascripts/application.js

```diff
@@ -104,4 +104,54 @@ function validateNewsLetterEmail(event) {
     $('#nl_message').html("");
     $('#nl_message').hide();
    }
+}
+
+function veriryNewsLetterEmailInSignUpPage(val, email, alt_email) {
+  // If alternate email is present in sign-up page take alternate-email to send news-letters other-wise DOE email.
+  news_letter_email = alt_email
+  if (news_letter_email == ""){
+    news_letter_email = email
+  }
+  $('#news-letter-ajax-message').show();
+  $('#news-letter-ajax-message').html('Verifying email')
+  $('#news-letter-ajax-message').css("color", '#c4262d');
+  $("#sign_up_button").prop("disabled", false);
+  $("#sign_up_button").css("background-color", "grey");
+
+  // Ajax call to validate the news-letter email.
+  $.ajax({ type: "POST", url: "/news_letter/validate_news_letter_email_from_user_sign_up_page", contentType: "application/json; charset=utf-8",
+           data: JSON.stringify({email: news_letter_email, sign_up_page: true}), dataType: "json", success: function (result) {
+
+    if ($('input[name=news_letter_email]').is(':checked')) {
+      if (result.success == "success") {
```

Comment:
What kinds of email validation are we doing?  Anything security-related?

Reply:
this is already existing new-letter validation only. Just checking the format of email and checking email is there in google sheets or not.